pkg/controllers/user/multiclusterapp/controller_multiclusterapp.go

```diff
@@ -93,6 +94,11 @@ func (m *MCAppController) sync(key string, mcapp *v3.MultiClusterApp) (runtime.O
 		return mcapp, err
 	}
 
+	mcapp = mcapp.DeepCopy()
+	if err := m.reconcileTargetsForDelete(mcapp); err != nil {
```

Comment:
just curious, why is deepcopy needed here?

Reply:
1) there's a possibility mcapp will change in cache with 
https://github.com/rancher/rancher/blob/c0d71e090a1ac7c0dbd575790092b04a9a6a2daf/pkg/controllers/user/multiclusterapp/controller_multiclusterapp.go#L485 

2) Not related to this PR but if `createApps` returns the object without Deepcopy, L117-147 will do the updates on actual object instead of copy https://github.com/rancher/rancher/blob/c0d71e090a1ac7c0dbd575790092b04a9a6a2daf/pkg/controllers/user/multiclusterapp/controller_multiclusterapp.go#L112